File: src/types/friction.ts
export type FrictionInsightSeverity = "high" | "medium" | "low";

export type FrictionInsightKind =
  | "project_switching"
  | "context_switching"
  | "support_mode"
  | "meeting_recovery_gap"
  | "stale_task"
  | "repeated_issue"
  | "late_report"
  | "focus_fragmentation";

export type FrictionInsightMetric = {
  key: string;
  label: string;
  value: string;
  unit?: string | null;
  threshold?: string | null;
  direction?: "above" | "below" | "equal" | string | null;
};

export type FrictionEvidenceSourceType =
  | "weekly_task"
  | "activity"
  | "manual_log"
  | "calendar_event"
  | "focus_session"
  | "report";

export type FrictionEvidenceItem = {
  evidenceId: string;
  sourceType: FrictionEvidenceSourceType;
  sourceId: string;
  title: string;
  date: string;
  occurredAt?: string | null;
  projectName?: string | null;
  detail?: string | null;
  route?: string | null;
  role?: "primary" | "supporting" | "counter" | string | null;
  observedValue?: string | null;
  routeState?: unknown;
};

export type FrictionInsightAction = {
  route: string;
  stateJson?: unknown;
  sourceId?: string | null;
};

export type FrictionInsightScope = {
  from: string;
  to: string;
  surface: string;
  projectIds?: string[] | null;
  classification?: string | null;
};

export type FrictionInsightClaim = {
  statement: string;
  impactLabel: FrictionInsightSeverity | string;
};

export type FrictionInsightDataHealth = {
  status: "complete" | "partial" | "limited" | string;
  notes: string[];
};

export type FrictionInsightReason = {
  id: string;
  label: string;
  detail: string;
  strength: "primary" | "supporting" | "limiting" | string;
  evidenceIds: string[];
};

export type FrictionInsight = {
  id: string;
  nudgeKey: string;
  ruleVersion: string;
  scope: FrictionInsightScope;
  claim: FrictionInsightClaim;
  kind: FrictionInsightKind;
  severity: FrictionInsightSeverity;
  confidence: number;
  confidenceLabel: "strong" | "likely" | "watch" | "needs_review" | string;
  verified: boolean;
  dataHealth: FrictionInsightDataHealth;
  title: string;
  detail: string;
  recommendation: string;
  evidence: string[];
  metrics: FrictionInsightMetric[];
  evidenceItems: FrictionEvidenceItem[];
  reasons: FrictionInsightReason[];
  actionLabel: string;
  actionTarget: string;
  primaryAction?: FrictionInsightAction | null;
  date?: string | null;
};

export type GetFrictionInsightsInput = {
  from: string;
  to: string;
  projectIds?: string[] | null;
  classification?: string | null;
  surface?: "today" | "dashboard" | "reports" | "weekly_plan" | "friction" | null;
};

File: src-tauri/src/domain/friction.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GetFrictionInsightsInput {
    pub from: String,
    pub to: String,
    pub project_ids: Option<Vec<String>>,
    pub classification: Option<String>,
    pub surface: Option<String>,
}

#[derive(Debug, Clone, Serialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum FrictionInsightSeverity {
    High,
    Medium,
    Low,
}

#[derive(Debug, Clone, Serialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum FrictionInsightKind {
    ProjectSwitching,
    ContextSwitching,
    SupportMode,
    MeetingRecoveryGap,
    StaleTask,
    RepeatedIssue,
    LateReport,
    FocusFragmentation,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FrictionInsightMetric {
    pub key: String,
    pub label: String,
    pub value: String,
    pub unit: Option<String>,
    pub threshold: Option<String>,
    pub direction: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "snake_case")]
pub enum FrictionEvidenceSourceType {
    WeeklyTask,
    Activity,
    ManualLog,
    CalendarEvent,
    FocusSession,
    Report,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FrictionEvidenceItem {
    pub evidence_id: String,
    pub source_type: FrictionEvidenceSourceType,
    pub source_id: String,
    pub title: String,
    pub date: String,
    pub occurred_at: Option<String>,
    pub project_name: Option<String>,
    pub detail: Option<String>,
    pub route: Option<String>,
    pub role: String,
    pub observed_value: Option<String>,
    pub route_state: Option<serde_json::Value>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FrictionInsightScope {
    pub from: String,
    pub to: String,
    pub surface: String,
    pub project_ids: Option<Vec<String>>,
    pub classification: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FrictionInsightClaim {
    pub statement: String,
    pub impact_label: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FrictionInsightDataHealth {
    pub status: String,
    pub notes: Vec<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FrictionInsightReason {
    pub id: String,
    pub label: String,
    pub detail: String,
    pub strength: String,
    pub evidence_ids: Vec<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FrictionInsightAction {
    pub route: String,
    pub state_json: Option<serde_json::Value>,
    pub source_id: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct FrictionInsight {
    pub id: String,
    pub nudge_key: String,
    pub rule_version: String,
    pub scope: FrictionInsightScope,
    pub claim: FrictionInsightClaim,
    pub kind: FrictionInsightKind,
    pub severity: FrictionInsightSeverity,
    pub confidence: f64,
    pub confidence_label: String,
    pub verified: bool,
    pub data_health: FrictionInsightDataHealth,
    pub title: String,
    pub detail: String,
    pub recommendation: String,
    pub evidence: Vec<String>,
    pub metrics: Vec<FrictionInsightMetric>,
    pub evidence_items: Vec<FrictionEvidenceItem>,
    pub reasons: Vec<FrictionInsightReason>,
    pub action_label: String,
    pub action_target: String,
    pub primary_action: Option<FrictionInsightAction>,
    pub date: Option<String>,
}
